src/transformers/models/whisper/generation_whisper.py

```diff
@@ -1167,49 +1190,47 @@ def replace_or_add(lst: List[int], num: int, itr: Iterator[int]):
                 num_segment_frames=num_segment_frames,
             )
 
-            if torch.unique(lang_ids).shape[0] > 1:
-                raise ValueError(
-                    "Multiple languages detected when trying to predict the most likely target language for transcription. It is currently not supported to transcribe to different languages in a single batch. Please make sure to either force a single language by passing `language='...'` or make sure all input audio is of the same language."
-                )
-
-            lang_id = lang_ids[0].item()
-
             # append or replace lang_id to init_tokens
-            if len(init_tokens) > 1:
-                init_tokens[1] = lang_id
-            else:
-                init_tokens.append(lang_id)
-
-        if task is not None:
-            if task in TASK_IDS:
-                init_tokens.append(generation_config.task_to_id[generation_config.task])
-                task_id = generation_config.task_to_id[generation_config.task]
+            for i in range(len(init_tokens)):
```

Comment:
The rest code assumes that we're using the same init tokens across the entire batch, but in the proposed changes we're looping over each batch item and computing the init tokens for each. This is redundant since the init tokens will be the same for each element of the batch. In practice, we should only need to compute them once!

Instead of doing this looping, can we keep the existing code and compute the init tokens just once? We can then copy them for each element in the batch, in a similar way to how you did previously:

```python
        # Separate init_tokens for each language
        init_tokens = [copy.deepcopy(init_tokens) for _ in language]
```

Reply:
This is already happening. `init_tokens` is of length 1, except if a list of languages is passed.

Reply:
The current version of the code makes it clear why we need to do this - happy with how it is! https://github.com/huggingface/transformers/pull/29688#discussion_r1559490951